packages/gatsby/src/services/start-webpack-server.ts

```diff
@@ -109,9 +112,45 @@ export async function startWebpackServer({
         webpackActivity.end()
         webpackActivity = null
       }
-      enqueueFlush()
+
+      if (isSuccessful) {
+        const state = store.getState()
+        const mapOfTemplatesToStaticQueryHashes = mapTemplatesToStaticQueryHashes(
+          state,
+          stats.compilation
+        )
+
+        mapOfTemplatesToStaticQueryHashes.forEach(
+          (staticQueryHashes, componentPath) => {
+            if (
+              !isEqual(
+                state.staticQueriesByTemplate.get(componentPath),
+                staticQueryHashes.map(String)
```

Comment:
We shouldn't map here, right? We should do it when we save identifiers.
```suggestion
                staticQueryHashes
```